src/emqx_auth_http.erl

```diff
@@ -35,28 +35,69 @@ init({AuthReq, SuperReq}) ->
-    case request(Method, Url, Params1) of
-        {ok, 200, _Body}  -> {ok, is_superuser(SuperReq, Client)};
+    IsSuperUser = fun() -> is_superuser(SuperReq, Client, Password) end,
+    auth_query(Method, Url, Params1, Headers, urlencoded, IsSuperUser).
+
+description() -> "Authentication by HTTP API".
+
+auth_query(Method, Url, Params, Headers, BodyType, SuperUser) ->
+    case request(Method, Url, Params, Headers, #{body_type => BodyType}) of
+        {ok, 200, _Body}  -> {ok, SuperUser()};
         {ok, Code, _Body} -> {error, Code};
         {error, Error}    -> lager:error("HTTP ~s Error: ~p", [Url, Error]),
                              {error, Error}
     end.
 
-description() -> "Authentication by HTTP API".
-
 %%--------------------------------------------------------------------
 %% Is Superuser?
 %%--------------------------------------------------------------------
 
--spec(is_superuser(undefined | #http_request{}, mqtt_client()) -> boolean()).
-is_superuser(undefined, _MqttClient) ->
+-spec(is_superuser(undefined | #http_request{}, mqtt_client(), Password :: binary()) -> boolean()).
+is_superuser(undefined, _MqttClient, _Password) ->
     false;
-is_superuser(#http_request{method = Method, url = Url, params = Params}, MqttClient) ->
-    case request(Method, Url, feedvar(Params, MqttClient)) of
+is_superuser(#http_request{method = Method, url = Url, params = _Params, headers = Headers, body_type = json}, MqttClient, Password) ->
+    case json_ril(MqttClient, Password) of
+        {error, Reason} ->
+            lager:error("HTTP auth failed, Url: ~s, Error: ~p", [Url, Reason]),
+            {error, Reason};
+        Body ->
+            is_superuser_query(Method, Url, Body, Headers, json)
+    end;
+is_superuser(#http_request{method = Method, url = Url, params = Params, headers = Headers, body_type = urlencoded}, MqttClient, _Password) ->
+    is_superuser_query(Method, Url, feedvar(Params, MqttClient), Headers, urlencoded).
+
+is_superuser_query(Method, Url, Params, Headers, BodyType) ->
+    case request(Method, Url, Params, Headers, #{body_type => BodyType}) of
         {ok, 200, _Body}   -> true;
         {ok, _Code, _Body} -> false;
         {error, Error}     -> lager:error("HTTP ~s Error: ~p", [Url, Error]), false
     end.
 
+json_ril(Client = #mqtt_client{username = Username}, Password) ->
+    try
+        [_ClusterID, IMEI] = binary:split(Username, <<"_">>),
+        jsx:encode(#{<<"deviceInfo">> => #{
+                        <<"info">> => #{
+                            <<"type">> => <<"android">>,
+                            <<"imei">> => IMEI
+                        },
+                        <<"consumptionDeviceName">> => <<"JIO_IOT">>,
+                        <<"jToken">> => Password
+                    }})
+    catch
+        _E:{badmatch, _} -> {error, {bad_username, {username, Username}}};
+        _E:{badarg, _} -> {error, {bad_client, Client}};
+        _E:Reason -> {error, Reason}
+    end.
```

Comment:
Not sure how to do this. I considered put this kind of json schema into configuration file, but it is still complex to use and parse.